### PR TITLE
fix: improve release notes

### PR DESCRIPTION
**Pull request by ckerr**
#### Description of Change

Addresses some of the release-notes script issues that we found during the 4.0.0 rollout:

* if we're making a stable release, omit notes that have already been released in older branches
* sniff semantic commit types from PR subjects, not just from commit messages
* don't honor unrecognized semantic commit types
* add an option to specify the version name to be shown in the notes

also, some housekeeping:

* Don't hardcode the release-notes' placeholder comment
* Update prepare-release.js to pass in the target version number

@codebytere @BinaryMuse 

#### Checklist

- [x] PR description included and stakeholders cc'd
- [x] `npm test` passes
- [x] PR title follows semantic [commit guidelines](https://github.com/electron/electron/blob/master/docs/development/pull-requests.md#commit-message-guidelines)
- [x] [PR release notes](https://github.com/electron/clerk/blob/master/README.md) describe the change in a way relevant to app developers, and are [capitalized, punctuated, and past tense](https://github.com/electron/clerk/blob/master/README.md#examples).

#### Release Notes

Notes: no-notes

### CODE REVIEW

File: script/release-notes/notes.js [hunk 1]
```diff
@@ -437,37 +438,81 @@ const getNotes = async (fromRef, toRef) => {
...
+
+    // if we still don't have a note, it's because someone missed a 'Notes:
+    // comment in a PR somewhere... use the PR subject as a fallback.
+    commit.note = commit.note || prSubject
   }
 
-  // remove uninteresting commits
+  // remove procedural commits
   pool.commits = pool.commits
     .filter(commit => commit.note !== NO_NOTES)
```
Comment (MarshallOfSound) [inline script/release-notes/notes.js]: Just noticed this here but there are a multitude of things that correspond to `No Notes`.

There is a complete list here --> https://github.com/electron/clerk/blob/master/src/index.ts#L4-L13

File: script/release-notes/notes.js [hunk 2]
```diff
@@ -437,37 +438,81 @@ const getNotes = async (fromRef, toRef) => {
...
+      pr = prParsed.pr && (prParsed.pr.number !== pr.number) ? prParsed.pr : null
     }
+
+    // if we still don't have a note, it's because someone missed a 'Notes:
+    // comment in a PR somewhere... use the PR subject as a fallback.
+    commit.note = commit.note || prSubject
   }
 
-  // remove uninteresting commits
+  // remove procedural commits
```
Comment (codebytere) [inline script/release-notes/notes.js]: nit: procedural is a bit vague: "non-user-facing" maybe?

File: script/release-notes/notes.js [hunk 3]
```diff
@@ -437,37 +438,81 @@ const getNotes = async (fromRef, toRef) => {
...
+          .reduce(reducer, null)
+        if (!semver.valid(earliestRelease)) {
+          return true
+        }
+        return semver.diff(earliestRelease, newVersion).includes('patch')
+      })
+  }
+
   const notes = {
     breaks: [],
```
Comment (codebytere) [inline script/release-notes/notes.js]: s/breaks/breaking?

Comment (ckerr) [reply script/release-notes/notes.js]: 7efd5f594c7d65fc6d32e28fd6af2ff9c4ea28b2

File: script/release-notes/notes.js (hunk 2 above)
Comment (ckerr) [reply script/release-notes/notes.js]: 4c6510811dea1f75507ab86d348a1f1b3d829fee

File: script/release-notes/notes.js (hunk 1 above)
Comment (ckerr) [reply script/release-notes/notes.js]: e5ee608d57b10fa91d695ea014973b5800ca2f2e

**Pull request merged**